tests/e2e/configurer/chain/queries.go

```diff
@@ -72,6 +75,41 @@ func (n *NodeConfig) QueryGRPCGateway(path string, parameters ...string) ([]byte
 	return bz, nil
 }
 
+func (n *NodeConfig) QueryNumPools() uint64 {
+	path := "osmosis/gamm/v1beta1/num_pools"
+
+	bz, err := n.QueryGRPCGateway(path)
+	require.NoError(n.t, err)
+
+	//nolint:staticcheck
+	var numPools gammtypes.QueryNumPoolsResponse
+	err = util.Cdc.UnmarshalJSON(bz, &numPools)
+	require.NoError(n.t, err)
+	return numPools.NumPools
+}
+
+func (n *NodeConfig) QueryConcentratedPool(poolId uint64) (cltypes.ConcentratedPoolExtension, error) {
+	path := fmt.Sprintf("/osmosis/concentratedliquidity/v1beta1/pools/%d", poolId)
+	bz, err := n.QueryGRPCGateway(path)
+	require.NoError(n.t, err)
+
+	var poolResponse cltypes.QueryPoolResponse
+	err = util.Cdc.UnmarshalJSON(bz, &poolResponse)
+	require.NoError(n.t, err)
+
+	var pool poolmanagertypes.PoolI
+	err = util.Cdc.UnpackAny(poolResponse.Pool, &pool)
+	require.NoError(n.t, err)
+
+	poolCLextension, ok := pool.(cltypes.ConcentratedPoolExtension)
+
+	if !ok {
+		return nil, fmt.Errorf("Could not typecast %#v to type ConcentratedPoolExtension", pool.String())
```

Comment:
nit: what I meant was something like this: https://github.com/osmosis-labs/osmosis/blob/801eadf4fe2be2f620b5c5594da8b7ed25e4ccac/x/mint/types/params.go#L152-L155